docs/proposals/0683-epp-architecture-proposal/interfaces/interface.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"context"
+	"sync"
+
+	scheduling "sigs.k8s.io/gateway-api-inference-extension/pkg/epp/scheduling/types"
+)
+
+// Plugin is the parent type for all the scheduling framework plugins.
+type Plugin interface {
+	Name() string
+}
+
+type Endpoint interface {
+	GetState() EndpointState
+	GetScore() float32
+	SetScore(val float32)
+}
+
+type EndpointState struct {
+	// only need to use a sync.Map if we do not plan on snapshotting data.
+	storage sync.Map
+}
+
+type SchedulingResult struct {
+	results map[string][]Endpoint
+}
+
+type Scheduler interface {
+	Plugin
+	// ProfileSelection selects scheduling profiles through the implemented
+	// logic, and returns a subset of the registered scheduling profiles.
+	ProfileSelection() map[string]SchedulingProfile
```

Comment:
This call should receive the request or some request properties in order to select the profiles (decision on which profiles should be based on the request props)